src/msgpack/MapConverter.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace TarantoolDnx.MsgPack
+{
+    internal class MapConverter<TMap, TKey, TValue> : MapConverterBase<TMap, TKey, TValue>
+        where TMap : IDictionary<TKey, TValue>
+    {
+        public override void Write(TMap value, Stream stream, MsgPackSettings settings)
+        {
+            if (value == null)
+            {
+                settings.NullConverter.Write(value, stream, settings);
+                return;
+            }
+
+            WriteMapHeaderAndLength(value.Count, stream);
+            var keyConverter = settings.GetConverter<TKey>();
+            var valueConverter = settings.GetConverter<TValue>();
+
+            ValidateConverters(keyConverter, valueConverter);
+
+            foreach (var element in value)
+            {
+                keyConverter.Write(element.Key, stream, settings);
+                valueConverter.Write(element.Value, stream, settings);
+            }
+        }
+
+        public override TMap Read(Stream stream, MsgPackSettings settings, Func<TMap> creator)
+        {
+            var type = (DataTypes) stream.ReadByte();
+
+            uint length;
+            if (TryGetLengthFromFixMap(type, out length))
+                return ReadMap(stream, settings, creator, length);
+
+            switch (type)
+            {
+                case DataTypes.Null:
+                    return default(TMap);
```

Comment:
You can't because there are no conversion between null and TMap. If TMap is a reference type, then default(TMap) will be null.